source/main/physics/BeamFactory.h

```diff
@@ -181,13 +167,16 @@ class BeamFactory : public StreamableFactory < BeamFactory, Beam >, public Zeroe
 
 protected:
 	
+	std::shared_future<void> thread_future;
+
 	bool thread_mode;
 	int num_cpu_cores;
 
 	Beam *trucks[MAX_TRUCKS];
 	int free_truck;
 	int previous_truck;
 	int current_truck;
+	int simulatedTruck;

```

Comment:
Ahem... What about `m_simulated_truck` instead of `simulatedTruck`?

I would understand omitting the `m_` for consistency, but camelCase? This still applies you know: http://ror.avrintech.net/rorwikibackup/index.php/Coding_style#The_template :smile: